cordformation/src/main/kotlin/net/corda/plugins/Node.kt

```diff
@@ -541,7 +540,14 @@ open class Node @Inject constructor(private val project: Project) {
         Files.createDirectories(configDir)
         for ((jarFile, config) in cordapps) {
             if (config == null) continue
-            val configFile = configDir.resolve("${removeExtension(jarFile.fileName.toString())}.conf")
+            val fileNameWithoutExtension = jarFile.fileName.toString().let {
+                val end = when(val dotIndex = it.lastIndexOf('.')) {
```

Comment:
Maybe better?
```kotlin
when(val dotIndex = it.lastIndexOf('.')) {
    -1 -> it
    else -> it.substring(0, dotIndex)
}
```
You only need to invoke `substring()` if the filename has an extension.

Reply:
makes total sense...